201-timeseriesinsights-environment-with-eventhub/azuredeploy.json

```diff
@@ -192,24 +205,47 @@
       "tags": "[variables('environmentTagsValue')]",
       "resources": [
         {
+          "condition": "[empty(parameters('existingEventHubResourceId'))]",
           "type": "eventsources",
-          "name": "[parameters('eventSourceName')]",
+          "name": "[if(empty(parameters('existingEventHubResourceId')), parameters('eventSourceName'), 'avoidDuplicateResourceNameWhenCreatingEventHub')]",
```

Comment:
since this resource is conditional, you don't need extra logic on the name property - just use whatever property you would use when it's being created

Reply:
Unfortunately, validation doesn't seem to understand conditionals and thinks both resources are going to be deployed with the template. Without this "if" hack, I get the following validation failure:
>Deployment template validation failed: 'The resource 'Microsoft.TimeSeriesInsights/environments/GEN-UNIQUE-environmentName/eventsources/GEN-UNIQUE-eventSourceName' at line '209' and column '14' is defined multiple times in a template.